### ISSUE TITLE
fix documentation for twisted.internet.reactor

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5660]
|<img alt="Dev0n's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Dev0n reported|
|-|-|
|Trac ID|trac#5660|
|Type|enhancement|
|Created|2012-04-28 18:30:18Z|
|Branch|https://github.com/twisted/twisted/tree/reactor-docs-5660|

It states the default reactor is based on select(2) when in fact it's platform dependent now.

Current reactor how to will also need updating: http://twistedmatrix.com/documents/current/core/howto/choosing-reactor.html#auto2

```
The select reactor is currently the default reactor on all platforms.
```

Attachments:

* [twisted.internet.reactor-doc.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/7a9/7a9fc32a4288045bf1dc227cad8ff40c7a7eaa22/6132a56a42cc676a9dc74103f3eed2ae2abe0769.patch) (1063 bytes) - added by Dev0n on 2012-04-28 18:30:44Z - 
* [twisted.internet.reactor-doc_incl-howto_v2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/7a9/7a9fc32a4288045bf1dc227cad8ff40c7a7eaa22/861fe302719e135c60c28911c37cb87bff74497d.patch) (2147 bytes) - added by Dev0n on 2012-04-30 16:01:31Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5660 5660
type__enhancement enhancement
reporter__Dev0n Dev0n
priority__lowest lowest
milestone__ 
branch__branches_reactor_docs_5660 branches/reactor-docs-5660
branch_author__Dev0n Dev0n
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1335637818000000 1335637818000000
changetime__1337035229000000 1337035229000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set owner to Dev0n|
|-|-|

Thanks for noticing this and for your patch, Dev0n.  Even this updated documentation is going to be out of date very shortly, though.  [#5478](https://github.com/twisted/twisted/issues/5478) will change the default on Linux.  Ideally, _that_ change would actually include documentation updates to keep things accurate.  We'll leave this ticket open for now though, and see what happens.

Apart from that, there is at least one other piece of documentation which also needs to be updated, the core "choosing reactor" howto, which also talks about defaults.

It's unfortunate that this information needs to be duplicated in three places (implementation of default selection, `twisted.internet.reactor` docstring, and howto).  That makes it a lot easier for some documentation to get out-dated.  If we could consolidate this somehow, it'd be pretty great.

Once we figure out what action to take on this and there's a new patch up for consideration attached to this ticket, make sure to attach the "review" keyword (as described on ReviewProcess) to ensure the change gets attention in a timely manner.  Thanks again!

**Comment by twisted-trac** [comment]
|<img alt="Dev0n's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|Dev0n commented|
|-|-|

Thanks exarkun. Looks like [#5478](https://github.com/twisted/twisted/issues/5478) is ready for review so like you said, this would update again very soon. I've also attached the new patch.

**Comment by twisted-trac** [comment]
|<img alt="Dev0n's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|Dev0n removed owner|
|-|-|

Read for review, new patch includes the changes in howto. 

```
twisted.internet.reactor-doc_incl-howto_v2.patch 
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst commented|
|-|-|

(In [34401]) Branching to 'reactor-docs-5660'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst commented|
|-|-|

Thanks for the patch; I will commit a slightly tweaked version.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_6"></a>|@itamarst set status to `closed`|
|-|-|

(In [34406]) Merge reactor-docs-5660: Better documentation on default reactor.

Author: Dev0n
Review: exarkun, itamar
Fixes: [#5660](https://github.com/twisted/twisted/issues/5660)
